src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java

```diff
@@ -95,48 +114,68 @@ public void refreshScanManagers() throws IOException {
         final Set<Path> paths = Sets.newHashSet();
         int projectHash = Utils.getProjectIdentifier(project);
         ScanManager scanManager = this.scanManagers.get(projectHash);
+        ScanLogic logic = createScanLogic();
         if (scanManager != null) {
             scanManagers.put(projectHash, scanManager);
         } else {
             // Unlike other scan managers whereby we create them if the package descriptor exist, the Maven
             // scan manager is created if the Maven plugin is installed and there are Maven projects loaded.
-            createScanManagerIfApplicable(scanManagers, projectHash, ScanManagerTypes.MAVEN, "");
+            createScanManagerIfApplicable(scanManagers, projectHash, ScanManagerTypes.MAVEN, "", logic);
         }
         paths.add(Utils.getProjectBasePath(project));
-        createScanManagers(scanManagers, paths);
-        createPypiScanManagerIfApplicable(scanManagers);
+        createScanManagers(scanManagers, paths, logic);
+        createPypiScanManagerIfApplicable(scanManagers, logic);
         this.scanManagers = scanManagers;
     }
 
-    private void createScanManagers(Map<Integer, ScanManager> scanManagers, Set<Path> paths) throws IOException {
+    private ScanLogic createScanLogic() {
+        Logger log =Logger.getInstance();
+        XrayClient client = createXrayClientBuilder(GlobalSettings.getInstance().getServerConfig(), log).build();
+        try {
+            XrayScanCache scanCache = new XrayScanCache(project.getName(), HOME_PATH.resolve("cache"), log);
+
+            if (GraphScanLogic.isXrayVersionSupported(client.system().version())) {
+                return new GraphScanLogic(scanCache,log);
+            } else {
+                if (BulkScanLogic.isXrayVersionSupported(client.system().version())) {
+                    return new BulkScanLogic(scanCache, log);
+                }
+            }
+        } catch (IOException e) {
```

Comment:
Better not to catch the exception, but let it be thrown. We should strive to catch them in upper most part of the stack trace as possible. The code is "healthier" this way.